image-viewer.py

```diff
@@ -55,5 +58,82 @@ def render(self):
+        menubar.add_cascade(label='File', menu=file_menu, underline=0) # Display the File button on the menu bar
+    
+    # Open the explorer and choose a new folder    
+    def open_folder(self):
+        new_dir = filedialog.askdirectory()
+        
+        # only change directory if a new one is selected
+        if new_dir != "": 
+            self.img_dir = new_dir
+            self.find_images(self.img_dir)
+    
+    # Refresh the current directory, incase images have been added or removed
+    def refresh(self):
+        self.find_images(self.img_dir)
+    
+    # Find images in input directory and append to image_files list    
+    def find_images(self, dir):
+        
+        # Clear current list of images and reset index
+        self.image_files = []
+        self.current_image_index = 0
+        
+        for item in listdir(dir):
+            if isfile(f"{dir}/{item}"):
+                # Only add image items with the following extensions
+                if(item.lower().endswith(('.png', '.jpg', '.jpeg', '.ico', '.gif'))):
+                    self.image_files.append(item)
+                
+        # Display
+        self.display_image()
+
+    # Display image at the current index
+    def display_image(self):
+        
+        try:
+            self.name_label.configure(text=self.image_files[self.current_image_index]) # Update the Label
+            new_img = Image.open(f"{self.img_dir}/{self.image_files[self.current_image_index]}") # Open the image
+            self.current_image = ImageTk.PhotoImage(self.resize_image(new_img)) # Resize and load the image
+            self.display.configure(image=self.current_image) # Display the image
+            
+        except OSError as e:
+            # No images were found - so change the label and display a blank image
+            self.name_label.configure(text=e)
+            self.current_image = ImageTk.PhotoImage(self.blank_img)
+            self.display.configure(image=self.current_image)
+            
+        except IndexError:
+            # No images were found - so change the label and display a blank image
+            self.name_label.configure(text="No image files found")
+            self.current_image = ImageTk.PhotoImage(self.blank_img)
+            self.display.configure(image=self.current_image)
+    
+    # Resize the images for display if they are bigger than 500 x 300 -- keeping their aspect ratio
+    def resize_image(self, image):
+        
+        if image.height > 300:
+            aspect_ratio = image.width / image.height
+            new_width = aspect_ratio * 300
+            return image.resize((int(new_width), 300), Image.Resampling.LANCZOS)
+        
+        elif image.width > 500:
+            aspect_ratio = image.width / image.height
+            new_height = 500 * aspect_ratio
+            return image.resize((500, int(new_height)), Image.Resampling.LANCZOS)
+        
+        else:
+            return image
```

Comment:
Conditionals can also be laid out like:
```
if image.height > 300:
    ...
    return image.resize(...)

if image.height > 500:
    ...
    return image.resize(...)

return image
```

I don't think it will make any difference performance-wise, but it can help with readability. In this case though, I wouldn't say this is problematic in the slightest, so I'll leave it up to you whether you want to make that change or not!